src/main/java/com/nasrpi/home/HomeRepository.java

```diff
@@ -160,4 +169,36 @@ public boolean moveFile(String source, String destination) {
 
 	}
 
+	public UploadFileResponse uploadFile(MultipartFile file, String uploadPath, FileStorageService fileStorageService) {
+
+		String fileName = fileStorageService.storeFile(file, uploadPath);
+
+		String fileDownloadUri = ServletUriComponentsBuilder.fromCurrentContextPath().path("/downloadFile/")
+				.path(fileName).toUriString();
+
+		return new UploadFileResponse(fileName, fileDownloadUri, file.getContentType(), file.getSize());
+	}
+
+	public ResponseEntity<Resource> downloadFile(String fileName, String filePath, HttpServletRequest request,
+			FileStorageService fileStorageService) {
+		// Load file as Resource
+		Resource resource = fileStorageService.loadFileAsResource(fileName, filePath);
+
+		// Try to determine file's content type
+		String contentType = null;
+		try {
+			contentType = request.getServletContext().getMimeType(resource.getFile().getAbsolutePath());
+		} catch (IOException ex) {
+		}
+
+		// Fallback to the default content type if type could not be determined
+		if (contentType == null) {
+			contentType = "application/octet-stream";
+		}
+
+		return ResponseEntity.ok().contentType(MediaType.parseMediaType(contentType))
+				.header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=\"" + resource.getFilename() + "\"")
```

Comment:
add strings to constants class and then use them

Reply:
done